meta.go

```diff
@@ -0,0 +1,100 @@
+package yarpc
+
+import (
+	"context"
+
+	"go.uber.org/yarpc/api/transport"
+)
+
+// CallOption defines options that may be passed in at call sites to other
+// services.
+//
+// These may be used to add or alter the request.
+type CallOption struct{ apply func(*Call) }
```

Comment:
I’m sure you considered applying directly to the *transport.Request. Is there a compelling reason for the indirection?